packages/shared/ReactTypes.js

```diff
@@ -179,6 +179,12 @@ export interface Thenable<+R> {
     onFulfill: (value: R) => void | Thenable<U> | U,
     onReject: (error: mixed) => void | Thenable<U> | U,
   ): void | Thenable<U>;
+
+  status?: 'pending' | 'fulfilled' | 'rejected';
+  // `value` is type R when fulfilled, and mixed when rejected.
+  // TODO: There's probably a correct way to model this in newer versions of
```

Comment:
We already model this pattern with enums like this: https://github.com/facebook/react/blob/main/packages/react-client/src/ReactFlightClient.js#L46-L81

The status field probably can't be optional but maybe lie about that?

Reply:
Managed to get it working